<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-0.9.649">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="William Pincombe">

<title>README</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>


<script src="README_files/libs/clipboard/clipboard.min.js"></script>
<script src="README_files/libs/quarto-html/quarto.js"></script>
<script src="README_files/libs/quarto-html/popper.min.js"></script>
<script src="README_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="README_files/libs/quarto-html/anchor.min.js"></script>
<link href="README_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="README_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="README_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="README_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="README_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet">

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">README</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>William Pincombe </p>
          </div>
  </div>
    
    
  </div>
  

</header>

<p>A guide document to everything done in the collaboration project with Karl Berator.</p>
<section id="raw-data" class="level1">
<h1>Raw Data</h1>
<p>The data file <code>WTF-IISfD data.xlsx</code> was received from Karl Berator through a Google Drive link on 1st March 2023. It is saved in the <code>raw-data</code> folder.</p>
</section>
<section id="cleaning-of-data" class="level1">
<h1>Cleaning of data</h1>
<p>Manually cleaned the data from Karl by copying into two new <code>.csv</code> files in LibreOffice Calc. These files are saved within a folder called <code>cleaned-data</code>.</p>
<ul>
<li><p>in <code>manually-cleaned-data-wide.csv</code>, the values of gene expression for each observation series are saved as a column, with the concentration as the independent variable. This file does not contain information about the gene line.</p></li>
<li><p>in <code>manually-cleaned-data-long.csv</code>, gene expression is only one variable, with each trial (for different treatment, gene type and gene line) as a separate set of rows. The details for each observation, treatment, gene type and gene line, are given by categorical variables.</p></li>
</ul>
<p>I have generally used <code>manually-cleaned-data-long.csv</code> in analysis.</p>
</section>
<section id="plots" class="level1">
<h1>Plots</h1>
<p><code>01_plots.R</code> in the <code>R</code> folder is a script for experimenting with plots.</p>
<p>Tables user the “wider” <code>manually-cleaned-data-wide.csv</code> whereas the plots use the “longer” <code>manually-cleaned-data-long.csv</code></p>
<p>This file does not contain the final version of these plots - that is in <code>slides.qmd</code> in the <code>slides</code> folder.</p>
</section>
<section id="deliverable-1-slides" class="level1">
<h1>Deliverable 1: Slides</h1>
<p>See the <code>slides</code> folder.</p>
<p>Slides were initially constructed using <code>slides.qmd</code> in main directory, to create the figures and tables. This was for implementing the plots from the 01_plots.R in a way that is exportable to PowerPoint (the desired format of Karl Berator).</p>
<p>The compiled output of <code>slides.qmd</code> was then edited manually using LibreOffice Impress (an open-source alternative for PowerPoint) and the result was saved in <code>Slides for Karl.pptx</code></p>
</section>
<section id="deliverable-2-plot" class="level1">
<h1>Deliverable 2: Plot</h1>
<p>See the <code>gene-plot</code> folder.</p>
<p>Received the plot to emulate, saved as <code>gene_plot.pdf</code> in folder <code>gene-plot</code>.</p>
<p>Write R script <code>gene_plot_updating.R</code> to create a version of the plot that has:</p>
<ul>
<li><p>Font: Times New Roman</p></li>
<li><p>.tiff format</p></li>
<li><p>(9in x 6in) with a resolution of 500</p></li>
</ul>
<p>Plot uses the same data we had previously looked at, imported from <code>manually-cleaned-data-long.csv</code>.</p>
<p>The Times New Roman font file is in the directory to be called on within the script.</p>
<p>The script saves the plot as a <code>.tiff</code> file with the specified characteristics, called <code>gene_plot_updated.tiff</code>.</p>
<p>So that this file could be sent via email, I manually (using the Linux Mint operating system) compressed it into a <code>.zip</code> file, called <code>gene_plot_updated.zip</code>.</p>
</section>
<section id="deliverable-3-sample-size-calculation" class="level1">
<h1>Deliverable 3: Sample size calculation</h1>
<p>R script to do the calculation is <code>03_regression_sample_size.R</code> in the <code>R</code> folder.</p>
<p>I calculated the sample size using the <code>pwr.f2.test</code> function from the <code>pwr</code> package. This function can calculate, for a linear model, one of the following from values of all four others:</p>
<ul>
<li><p>numerator degrees of freedom <span class="math inline">\(u\)</span></p></li>
<li><p>denominator degrees of freedom <span class="math inline">\(v\)</span></p></li>
<li><p>effect size</p></li>
<li><p>significance level</p></li>
<li><p>power</p></li>
</ul>
<p>Karl specified in his email that he wants power of 90% and a significance level of 0.05.</p>
<p>He also specified that he had an <span class="math inline">\(R^2\)</span> of 0.1. We can calculate the effect size <span class="math inline">\(f^2\)</span> from this using the formula which can be found in <em>Sample Size Determination and Power</em> by Thomas P. Ryan, 2013, on page 149:</p>
<p><span class="math display">\[
f^2 = \frac{R^2}{1-R^2}
\]</span></p>
<p>The function is doing an F-test, so if we have <span class="math inline">\(k\)</span> parameters and <span class="math inline">\(n\)</span> observations, the degrees of freedom are <span class="math inline">\(u = k\)</span> and <span class="math inline">\(v = n - k - 1\)</span>. The number of parameters counts:</p>
<ul>
<li><p>each continuous predictor once</p></li>
<li><p>each factor variable the number of levels minus 1, for the base level</p></li>
</ul>
<p>In Karl’s case, there are two continuous predictors (concentration and cell age) and three factor predictors (treatment, cell type and media), each with 2 levels, so we have <span class="math inline">\(k = 2+3\times (2-1) = 5\)</span> predictors, so we know that <span class="math inline">\(u = 5\)</span> and <span class="math inline">\(v = n - 6\)</span>.</p>
<p>We then input the power, significance level, effect size and numerator degrees of freedom <span class="math inline">\(u\)</span> into the function to calculate the required denominator degrees of freedom <span class="math inline">\(v\)</span>, and then let <span class="math inline">\(n = v + 6\)</span>.</p>
</section>
<section id="modelling" class="level1">
<h1>Modelling</h1>
<section id="files" class="level2">
<h2 class="anchored" data-anchor-id="files">Files</h2>
<p>The model fitting and analysis is split over two files:</p>
<ul>
<li><p><code>models.qmd</code> contains analysis that was not necessarily included in the final report.</p></li>
<li><p><code>imrad.qmd</code> in the <code>IMRaD-report</code> folder contains the final analysis and report that was sent to Karl Berator</p></li>
</ul>
<p><code>imrad.qmd</code> has the more up-to-date version of any analysis that appears in both.</p>
</section>
<section id="modelling-process" class="level2">
<h2 class="anchored" data-anchor-id="modelling-process">Modelling process</h2>
<p>Our goal in modelling was to analyse at the relationship between gene expression and concentration, and how this was impacted by the treatment.</p>
<p>I used the “long” format data for modelling, so we have five variables:</p>
<ul>
<li><p><code>Gene Expression</code> is a continuous variable.</p></li>
<li><p><code>Concentration</code> of growth factor, measured in <span class="math inline">\(\mu\)</span>g/ml, is a continuous variable taking integer values between 0 and 11.</p></li>
<li><p><code>Treatment</code> is a factor variable with levels AF42 and placebo.</p></li>
<li><p><code>Cell Type</code> is a factor variable with levels WT and CT101.</p></li>
<li><p><code>Gene Line</code> is a factor variable with 8 levels: CsE, bNo, JZC, fUg, jEK, Hoe, Rza and xpo.</p></li>
</ul>
<p>I firstly did Exploratory Data Analysis on the data, in the <code>models.qmd</code> file, plotting</p>
<ul>
<li><p>a histogram of the response variable (gene expression), which shows that it is somewhat right-skewed, which could present a problem for our analysis due to the assumption of normality (i.e.&nbsp;no skewness) of the residuals.</p></li>
<li><p>scatter plots of the two continuous variables, gene expression and concentration, to look at</p>
<ul>
<li>the relationship between them</li>
<li>how this relationship differs across different levels of treatment and cell type</li>
</ul></li>
</ul>
<p>I initially fitted a simple linear model <code>M1</code> using concentration, treatment and cell type as predictors for gene expression, including interaction terms between treatment and concentration and between cell type and concentration.</p>
<ul>
<li><p>In <code>models.qmd</code>, I experimented with applying a log transformation on <code>M1</code> in order to</p></li>
<li><p>However, I could not find a consensus in the online literature about how to use a log transform in the mixed-effects setting, so since I could not compare this model to a mixed-effects equivalent, I abandoned this line of analysis.</p></li>
</ul>
<p>I then fit a mixed-effects model <code>M2</code>, with the same predictors as the fixed effects, including the interaction terms, with gene line as an intercept-only random effect.</p>
<p>I then fit mixed effects models <code>M3</code> and <code>M4</code> removing treatment and cell type respectively, to compare the model metrics (AIC and BIC) with those in <code>M2</code> to measure the significance of these variables.</p>
<p>As can be seen in the IMRaD Report, the AIC and BIC values for the complete mixed-effects model <code>M2</code> were much better than those of either the reduced models, or the simple linear model <code>M1</code> without the random effect. Hence, I decided to use <code>M2</code> as the final model.</p>
<p>Finally, I experimented with a random slope model into a random intercept and random slope model, but was informed that this level of complexity was not necessary for the report.</p>
</section>
<section id="assessing-models" class="level2">
<h2 class="anchored" data-anchor-id="assessing-models">Assessing Models</h2>
<p>In the IMRaD report, I produced diagnostic plots of <code>M2</code> in the IMRaD Report, showing</p>
<ul>
<li><p>a histogram of the residuals, to check residual normality</p></li>
<li><p>a scatter plot of the residuals against the fitted values, to check homoskedasticity and linearity</p></li>
</ul>
<p>I also plotted the fitted regression lines for each gene line over the scatter plot of the data, to check how well the model fitted the data.</p>
</section>
</section>
<section id="deliverable-4-imrad-report" class="level1">
<h1>Deliverable 4: IMRaD Report</h1>
<p>The IMRaD report is in a folder called <code>IMRaD-report</code>.</p>
<p>The report is a Quarto markdown document, <code>imrad.qmd</code>, which calls on a bibliography in BibTeX format <code>imrad.bib</code>.</p>
<p>The Times New Roman font file is also saved in this directory so that it can be used as the font in the plots in the report.</p>
</section>
<section id="miscellaneous" class="level1">
<h1>Miscellaneous</h1>
<p>In the top-level directory:</p>
<ul>
<li><code>Important Emails from Karl</code> is a text file containing some emails from Karl Berator which specified the tasks - so that I could look over more easily</li>
</ul>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>